guardian/src/main/java/com/auth0/android/guardian/sdk/Guardian.java

```diff
@@ -86,7 +86,7 @@
     @NonNull
     public GuardianAPIRequest<Void> delete(@NonNull Enrollment enrollment) {
         return client
-                .device(enrollment.getId(), enrollment.getDeviceToken())
+                .device(enrollment.getId(), enrollment.getUserId(), enrollment.getSigningKey())
```

Comment:
That would be simpler but the way the SDK is structured today is sticking to the _principle of least knowledge_. In this particular case, that means that the `Guardian API Client` class doesn't have to know anything about the `Enrollment` class.

Reply:
I don't think that this applies here. We are using a deconstructed enrollment instead of using the actual dataobject, in terms of what an enrollment is about all the information.
I guess that the tradeoff here how strong use of the java typessystem we want to do.

Reply:
@joseluisdiaz How I see this SDK, has two main classes for interacting with the API.
- The raw/base API client, called `GuardianAPIClient` which only takes Strings and Private Keys (JDK classes) and makes the requests. Source [here](https://github.com/auth0/Guardian.Android/blob/master/guardian/src/main/java/com/auth0/android/guardian/sdk/GuardianAPIClient.java).
- The wrapper, called `Guardian`, documented as the entry point for consumption of this SDK in the [readme](https://github.com/auth0/Guardian.Android#usage). This one takes DTOs, as you suggest above. Source [here](https://github.com/auth0/Guardian.Android/blob/master/guardian/src/main/java/com/auth0/android/guardian/sdk/Guardian.java).

The method signature under discussion here belongs to the `GuardianAPIClient` class. I'm saying that the class, to be consistent with what's already there, should keep accepting strings instead of being passed the `Enrollment` instance. Moreover, from that instance, we only require 3 properties of the [almost 10 defined](https://github.com/auth0/Guardian.Android/blob/master/guardian/src/main/java/com/auth0/android/guardian/sdk/Enrollment.java). IMHO passing the entire object just to pick these 3 properties is overkill.

In the end, this is not different from how it's [implemented in the Swift SDK](https://github.com/auth0/Guardian.swift/pull/91/files#diff-821395ed095e50eab6c2a7a88f087b284702d8949866d983823015f04acec2acR59). If you still think this implementation should be changed, please let me know what changes you want and I'll make them.

Reply:
I see your point. I'm concerted about who would create the token actually. This seems to be a good enough approach.